h5py/h5t.pyx

```diff
@@ -1793,10 +1793,12 @@ def opaque_dtype(np_dtype):
     the opaque data as datetimes.
     """
     dt = dtype(np_dtype)
-    if np.issubdtype(dt, np.object_):
+    if dt.hasobject:
         raise TypeError("Cannot store numpy object arrays as opaque data")
     if dt.names is not None:
         raise TypeError("Cannot store numpy structured arrays as opaque data")
+    if dt.subdtype is not None:
```

Comment:
Is there a reason this is blocked?

Reply:
I realised it wouldn't work as expected with subarrays - see my other comment.